feat: UTC-300: Configurable permission for managing user personal token

mcanu opened the pull request:
<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

### Backend:
- New `users_token_any` controls what users can access Legacy and JWT tokens
- Removed inconsistent permission checking for JWTSettings. Now doing it how it works across the codebase, by checking permissions in the API based on request method.
- Added `permission_required` to JWT token views that were only checking if the user was authenticated, but not the permission.

mcanu committed:
Added new permission users_token_any

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/18041315276)

mcanu committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18041315276

mcanu committed:
Merge branch 'develop' into 'fb-utc-300'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18041315276

mcanu committed:
Fixed permissions for JWT

mcanu committed:
Removed irrelevant tests

mcanu committed:
Removed permissions for refresh endpoint

nass600 committed:
added new authprovider

nass600 committed:
fixed typings

bmartel commented on web/apps/labelstudio/src/app/App.jsx:
```diff
@@ -70,6 +71,7 @@ const App = ({ content }) => {
             <ConfigProvider key="config" />,
             <RoutesProvider key="rotes" />,
             <ProjectProvider key="project" />,
+            <AuthProvider key="auth" />,
```
This should be put in the outermost layer of the providers, just below JotaiProvider, as it is possible to need access to its values from one of the other providers in the future, and it is only itself dependent on QueryClientProvider and JotaiProvider from what I can see.

nass600 replied:
Good observation, thanks

nass600 committed:
promoting authprovider scope

nass600 committed:
auth fixes

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/18133474297)

mcanu committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18133474297

mcanu committed:
Merge branch 'develop' into 'fb-utc-300'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18133474297

mcanu committed:
Reset poetry.lock

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/18139982858)

mcanu committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18139982858
